packages/kbn-expandable-flyout/src/utils/helpers.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FlyoutLayout } from '../models/layout';
+import { FlyoutPanel } from '../models/panel';
+import { FlyoutState } from '../models/state';
+
+/**
+ * Open the flyout by removing all prior left, right and preview panel and replacing with new ones.
+ *
+ * @param {FlyoutState} state the previous state
+ * @param payload optional left, right and/or preview panels
+ * @return {FlyoutState}
+ */
+export const openFlyout = (
```

Comment:
wait so this is an `actionCreator` of sorts? maybe it should not be named `helpers.ts` :)

Reply:
yup, as I was replacing `useState` with `useReducer` I went with a proper reducer/action setup, it's much nicer!